45 b61 shuttle integration, shuttle schedule

nicpuertas opened the pull request:
Added homescreen button to see shuttle schedule that redirects to Concordia shuttle schedule webpage. Campus Map and Directions Screens have shuttle station and buses icons that appear on the map. Buses location update every 15 seconds. Toggle feature to show the shuttle elements on/off. Quick testing shows everything working properly.

nicpuertas commented:
HomeScreen test fails because a shuttle schedule button was added

YFrancis10 commented:
The button in the home screen for the shuttle schedule works well. Although, when trying the to use the shuttle in the directions screen, I get this error: **(NOBRIDGE) ERROR  Warning: Text strings must be rendered within a <Text> component.** 

Also, I cannot get directions to a location by using the shuttle bus. It only gives me the directions for the metro and bus. Another thing is that the shuttle Icons on the map are pretty big and create a cluster. 

![sc directions](https://github.com/user-attachments/assets/010d471c-1780-4799-862b-8f53e7449b7b)

nicpuertas commented:
Error has been fixed and icons have been changed and adjusted to a more reasonable size. Being able to use the shuttle in directions searching will be implemented as part of B4 and B5. Currently working on it. This PR focused on implementing the shuttle functionality core before implementing it with our other features.

nicpuertas commented:
<img width="369" alt="image" src="https://github.com/user-attachments/assets/cb98d14a-022d-48b5-b8a4-db00574d4329" />
This feature identifies is the user is inside a concordia building. Clicking use my location will check and alert the user that he is in a building. Notice the Notification at the top left as well. Fetching directions using their location will adjust accordingly based on the building center.

nicpuertas commented:
<img width="368" alt="image" src="https://github.com/user-attachments/assets/b43ed636-f3e9-41dd-b268-143ab7fa2e20" />
<img width="387" alt="image" src="https://github.com/user-attachments/assets/b55880e6-6a23-49cb-80d3-ee781dfba1f7" />

Expansible/Collapsible directions with a shuttle option tab if available. Shuttle option indicated estimated time as well as a quick link to see the shuttle schedule. Can scroll in box to see directions and route details.

nicpuertas commented:
<img width="364" alt="image" src="https://github.com/user-attachments/assets/19fa5e5a-3cbe-452a-b2b4-2f402a0c311e" />
<img width="368" alt="image" src="https://github.com/user-attachments/assets/50e502f8-58dd-4f7a-8858-5e5e9cdb42f9" />
<img width="369" alt="image" src="https://github.com/user-attachments/assets/7acf74c1-52bc-4792-9297-ed4300443e20" />

This push allows user to use the buttons to set their origin/destination and system gives user feedback. Added a button so the user can clear the set points. Also added  button to go back to search so a user can perform another directions search instead of having to go back to home and back into directions screen.

nicpuertas commented:
Codecov tests fail for campusMap displaying info on marker press, but tested it on app and it works properly. Could it be the tests that are outdated? @mcndl13

mcndl13 commented:
> Codecov tests fail for campusMap displaying info on marker press, but tested it on app and it works properly. Could it be the tests that are outdated? @mcndl13 Gonna wait for confirmation before merging, we're going to merge this one first before merging accessibility PR and resolve conflicts from there.

Yes the test are outdated. After your merge I will work on the test.

mcndl13 reviewed (commented):
You guys did an amazing job. Most impressive PR so far.

Ready to merge!
